PocketKit/Sources/SaveToPocketKit/SavedItem/SavedItemViewModel.swift

```diff
@@ -63,18 +63,48 @@ class SavedItemViewModel {
     func save(from context: ExtensionContext?) async {
         guard appSession.currentSession != nil else {
             autodismiss(from: context)
+
+            return
+        }
+
+        guard let extensionItems = context?.extensionItems else {
+            tracker.track(event: Events.SaveTo.unableToSave())
+            autodismiss(from: context)
+
+            return
+        }
+
+        let urlsToSave = await parse(extensionItems: extensionItems)
+
+        if urlsToSave.isEmpty {
+            tracker.track(event: Events.SaveTo.unableToSave())
+            infoViewModel = .error
+            autodismiss(from: context)
+
             return
         }
 
-        let extensionItems = context?.extensionItems ?? []
+        save(urlsToSave)
+
+        autodismiss(from: context)
+    }
+
+    private func parse(extensionItems: [ExtensionItem]) async -> [String] {
+        var urlsToSave: [String] = []
 
         for item in extensionItems {
             guard let url = try? await url(from: item) else {
-                tracker.track(event: Events.SaveTo.unableToSave())
-                infoViewModel = .error
-                break
+                continue
             }
 
+            urlsToSave.append(url)
+        }
+
+        return urlsToSave
+    }
```

Comment:
If I'm interpreting the code correctly, we don't want to save _all_ URLs that are found; at least, that's not what the legacy app did, and not what the original objective of this share extension was / is. We want to find the _first_ URL that we can save, and save that.

Reply:
We only seem to pull one URL out, if we do pick one would you prefer we .first or .last?

Reply:
What the original code was _supposed_ to do, which the `break`s broke, was for each extension item, grab its url, if it could, and attempt to save the first URL that was validly pulled from an extension item. 

So, for the PDF-situation specifically, we had two extension items -- one whose UTI was `com.adobe.pdf`, and one whose UTI was `public.url`.  The `for` loop would attempt to extract a URL from the first extension item, and fail since its UTI wasn't correct, then move onto the second extension item, from which the URL _would_ be extracted, and we would then save that.

In a hypothetical world where there was a _third_ extension item, the fact that we pulled a URL to save from the _second_ extension item would stop the `for` loop, and the third extension item, no matter the URL, would not attempt to be saved.

TL;DR - Stop and save the _first_ extracted URL from an extension item, and only that URL.